esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -441,41 +496,43 @@ def main(cfg):
         obs_data, obs_data_files = read_observation_data(datasets_obs)
         obs_data = aggregate_obs_data(obs_data, operator='median')
 
-        if datasets_model[0].get('independence', True):
-            logger.info('Calculating independence for %s', variable_group)
-            independence = calculate_independence(model_data)
-            visualize_and_save_independence(independence, cfg,
-                                            model_data_files)
-            logger.debug(independence.values)
-            independences[variable_group] = independence
-
-        if datasets_model[0].get('performance', True):
-            logger.info('Calculating performance for %s', variable_group)
-            performance = calculate_performance(model_data, obs_data)
-            visualize_and_save_performance(performance, cfg,
-                                           model_data_files + obs_data_files)
-            logger.debug(performance.values)
-            performances[variable_group] = performance
-            obs_ancestors.extend(obs_data_files)
+        logger.info('Calculating performance for %s', variable_group)
+        performance = calculate_performance(model_data, obs_data)
+        visualize_and_save_performance(performance, cfg,
+                                       model_data_files + obs_data_files)
+        logger.debug(performance.values)
+        performances[variable_group] = performance
+        obs_ancestors.extend(obs_data_files)
 
         model_ancestors.extend(model_data_files)
 
-    logger.info('Computing overall mean independence')
-    independence = xr.Dataset(independences)
-    overall_independence = compute_overall_mean(independence)
-    visualize_and_save_independence(overall_independence, cfg, model_ancestors)
+    model_ancestors = list(set(model_ancestors))  # only keep unique items
 
-    logger.info('Computing overall mean performance')
-    performance = xr.Dataset(performances)
-    overall_performance = compute_overall_mean(performance)
-    visualize_and_save_performance(overall_performance, cfg,
-                                   model_ancestors + obs_ancestors)
+    if len(independences) > 0:
+        logger.info('Computing overall mean independence')
+        independence = xr.Dataset(independences)
+        overall_independence = compute_overall_mean(
+            independence, cfg['independence_contributions'])
```

Comment:
And this is where you won't have to parse it a second time

```suggestion
            independence, independence_contributions
```